### DISCUSSION TITLE
TFLite file larger than expected

### DISCUSSION THREAD

**Discussion by andynader** [larq/larq#696] [Q&A] [answered]
Hi, When I train the following architecture: 

    kwargs = {
        "input_quantizer": "ste_sign",
        "kernel_quantizer": "ste_sign",
        "kernel_constraint": "weight_clip",
    }

    binary_net = tf.keras.Sequential(
        [
            tf.keras.layers.Input(shape=(x_train.shape[1], x_train.shape[2])),
            lq.layers.QuantConv1D(32, (3,), padding="same", use_bias=False, **kwargs),
            lq.layers.QuantConv1D(32, (3,), padding="same",  use_bias=False, **kwargs),
            tf.keras.layers.MaxPooling1D(pool_size=2),
            lq.layers.QuantConv1D(64, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(64, (3,), padding="same", use_bias=False, **kwargs),
            tf.keras.layers.MaxPooling1D(pool_size=4),
            lq.layers.QuantConv1D(128, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(128, (3,), padding="same",  use_bias=False, **kwargs),
            tf.keras.layers.MaxPooling1D(pool_size=6),
            lq.layers.QuantConv1D(256, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(256, (3,), padding="same", use_bias=False, **kwargs),
            tf.keras.layers.MaxPooling1D(pool_size=6),
            lq.layers.QuantConv1D(256, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(256, (3,), padding="same", use_bias=False, **kwargs),
            tf.keras.layers.UpSampling1D(size=6),
            lq.layers.QuantConv1D(128, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(128, (3,), padding="same",  use_bias=False, **kwargs),
            tf.keras.layers.UpSampling1D(size=6),
            lq.layers.QuantConv1D(64, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(64, (3,), padding="same",  use_bias=False, **kwargs),
            tf.keras.layers.UpSampling1D(size=4),
            lq.layers.QuantConv1D(32, (3,), padding="same", use_bias=False, **kwargs),
            lq.layers.QuantConv1D(32, (3,), padding="same", use_bias=False, **kwargs),
            tf.keras.layers.UpSampling1D(size=2),
            lq.layers.QuantConv1D(3, (3,), padding="same",  use_bias=False, **kwargs),
            lq.layers.QuantConv1D(3, (3,), padding="same",  use_bias=False, **kwargs),
            tf.keras.layers.Flatten(),
            lq.layers.QuantDense(288, use_bias=False, **kwargs)
        ]
    )

lq.models.summary() suggests that the model should be around 150KB, yet when I convert it to a TFLite file using lce.convert_keras_model(), the file ends up being 5MB, which is the size of the floating point equivalent of the network. In addition, when I benchmark this model on a Pi 4, it ends up being equivalent in inference time to a full precision network. Any idea why this is happening?

**Comment by Tombana** [comment, marked as answer]
Hi @andynader,

Our tflite support currently includes 2-dimensional binary convolutions but not 1-dimensional ones. It should work if you use a 2D convolution but with height (or width) set to `1` and filter size `(3,1)` (or `(1,3)`). The `UpSampling1D` layers become `UpSampling2D` with `size=(1,x)`.

**Comment by andynader** [comment]
Yup, that worked, thank you so much!
